src/react-table-config.d.ts

```diff
@@ -0,0 +1,128 @@
+// check @types/react-table README for details on this config file:
```

Comment:
@bartaz is there a reason we can't just use `yarn install @types/react-table`? This seems a little unusual.

Reply:
@squidsoup we do, this config is 'part' of react-table types (it imports and extends types from @types/react-table).

I agree it's really strange and confusing (especially when I don't really know TS very well).
The react-table API (the shape of `data` and `columns` input objects) is very extensible. Depending on the features you use, you will add additional fields to the data and columns objects. So, to my understanding, @types/react-table defines the shape of these as whatever the defaults are in react-table, but you can use this config to turn on/off any features you use to make sure you only allow props that you need.

In our case, I want to pass `className` as a column prop. This is not something that is in react-table default props, so the native types don't support it and TS complains when sees `className` used.
So I used this config to add `className` to the `ColumnInstance` interface.

I don't know if that's the best way to do it, or can we somehow to these custom props somewhere else in the component itself?

The way this config file works is mystery to me. It seems to import a lot of things that are not used in the file, but if I don't import, the react-table module stops working, so I'm kind of trying to figure out what is needed here, what is not.

Personally I agree that this is quite confusing and unusual and would like to avoid using this file if we can find another way to extend types from @types/react-table.

Reply:
Managed to clean up this config to a reasonable minimum.